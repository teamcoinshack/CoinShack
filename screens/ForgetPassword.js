import React, { Component } from 'react'
import {
  Text, 
  View, 
  StyleSheet, 
  Dimensions,
  Alert,
} from 'react-native';
import Firebase from 'firebase';
import MyInput from "../components/MyInput.js";
import db from '../Database.js';

const background = '#373b48';

export default class ForgetPassword extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      username: '',
    }
    this.resetPassword = this.resetPassword.bind(this);
  }

  async resetPassword() {
    try {
      const isEmail = this.state.username.includes('@');
      const email = await db.searchExact(this.state.username, isEmail);
      if (email) {
        await Firebase.auth()
                .sendPasswordResetEmail(email);
        alert('Check your email to reset your password');
        this.props.navigation.navigate('Login');
      } else {
        alert('Unable to find user');
      }
    } catch(error) {
      console.log(error);
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={{ marginBottom: 20 }}>
          <Text style={{
            fontSize: 25,
            color: '#ffffff',
            fontWeight: 'bold'
          }}>
            Forgot your password? 
            </Text>
        </View>
        {this.state.errorMessage &&
          <Text style={{ color: 'red' }}>
            {this.state.errorMessage}
          </Text>}

        <View style={{ alignItems: "center" }}>
          <MyInput
            placeholder="Username/Email"
            onChangeText={username => this.setState({ username })}
            value={this.state.username}
            leftIconName="emoticon-outline"
          />
        </View>

        <MyButton
          text="Reset My Password"
          onPress={this.resetPassword}
          textColor="#00f9ff"
          width={Math.round(Dimensions.get('window').width) * 0.7}
        />

        <MyButton
          text="Back to Login"
          onPress={() => this.props.navigation.navigate('Login')}
          textColor="#00f9ff"
          width={Math.round(Dimensions.get('window').width) * 0.7}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: background,
  },
})
