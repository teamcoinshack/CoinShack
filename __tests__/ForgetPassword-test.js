/**
 * @format
 */

import 'react-native';
import React from 'react';
import ForgetPassword from '../screens/ForgetPassword.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native-google-signin', () => {});
test('renders correctly', () => {
  const tree = renderer.create(<ForgetPassword />).toJSON();
  expect(tree).toMatchSnapshot();
});
