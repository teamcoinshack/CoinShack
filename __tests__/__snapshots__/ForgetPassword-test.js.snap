// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#373b48",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, .4)",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="RNE__Overlay__backdrop"
    />
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#373b48",
            "borderRadius": 5,
            "height": 250,
            "padding": 10,
            "shadowColor": "rgba(0, 0, 0, .3)",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowRadius": 4,
            "width": 670,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#dbdbdb",
                "fontSize": 22,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            Error
          </Text>
          <Text
            style={
              Object {
                "color": "#a4a9b9",
                "fontSize": 20,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "height": 65,
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#515360",
                  "borderRadius": 5,
                  "elevation": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 10,
                  "marginLeft": 14,
                  "marginRight": 14,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ff077a",
                      "fontSize": 20,
                      "fontWeight": "700",
                    }
                  }
                >
                  OK
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Forgot your password?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 2,
            "borderColor": "#515360",
            "borderRadius": 5,
            "borderWidth": 2,
            "flexDirection": "row",
            "marginVertical": 10,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 525,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 15,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#999999",
                  "fontSize": 26,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï‡²
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Username/Email"
          placeholderTextColor="#999999"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "flex": 1,
              "fontSize": 20,
              "minHeight": 40,
              "padding": 10,
              "textAlign": "left",
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 65,
        "width": 525,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#515360",
          "borderRadius": 5,
          "elevation": 1,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 10,
          "marginLeft": 14,
          "marginRight": 14,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#00f9ff",
              "fontSize": 20,
              "fontWeight": "700",
            }
          }
        >
          Reset My Password
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 65,
        "width": 525,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#515360",
          "borderRadius": 5,
          "elevation": 1,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 10,
          "marginLeft": 14,
          "marginRight": 14,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#00f9ff",
              "fontSize": 20,
              "fontWeight": "700",
            }
          }
        >
          Back to Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
